Feature/only office (#151)
exoplatform/exo-documentation: docs/OnlyOffice.rst

.. _OnlyOffice:

#############################
OnlyOffice Connector for eXo
#############################

**What is OnlyOffice**

`OnlyOffice Online Editors <https://www.onlyoffice.com/office-suite.aspx>`__ is a great open source solution 

> that lets you to create, edit  and collaborate online on your office documents. 

Reviewer: ```suggestion
that lets you create, edit and collaborate online on your Office documents. 
```

To install the add-on, use the following command:

::

		./addon install exo-onlyoffice
		

.. _HowToUse:

================================
How to use OnlyOffice for eXo?
================================	

.. _EditOnline_BTN:

Edit Online button
~~~~~~~~~~~~~~~~~~~~
		
Having satisfied the above conditons and installed the add-on, you can easily edit your Microsoft Office documents online.
Just hit the ``Edit Online`` button |image1|.

The ``Edit Online`` button is available in these locations:

-  In the activity stream among the possible actions available on the document

   |image2|
   
-  In the document preview interface

   |image3|

-  In the ``Documents`` application, in the action bar

   |image4|
   
Clicking on this button loads the dcument in the corresponding OnlyOffice editor in new tab for maximum comfort.

.. _EditorInterface:

Editor Interface
~~~~~~~~~~~~~~~~~~

Clicking on |image5| button opens the document you want to edit in an another tab.
The interface contains the content of the document loaded in the OnlyOffice editor with a floating button.

> Clicking on that button opens a drawer "Save version" with these fields:

Reviewer: Is "drawer" a understandable term for everyone ?

Author: I think yes! any other proposals ?

Reviewer: If you think it is understandable, that's fine for me :) Otherwise, we could use "panel" maybe.

-  The file's breadcrumb
-  An edit icon allowing to edit the file's name
-  The summary text field

> -  Save button allow to add a version of the document in Documents application.

Reviewer: ```suggestion
-  A ``Save`` button allowing to create a new version of the document .
```

> -  List of the last previous 3 versions

Reviewer: ```suggestion
-  The list of the 3 latest versions
```

> -  See more button

Reviewer: ```suggestion
-  A ``See more`` button allowing to display the full version history of the document
```

-  A close button at the top right of the drawer

|image6|

File's breadcrumb
------------------

The file's breadcrumb contains:

> -  The space's avatar if the file is on a space or the user's avatar if the file is on Personal documents 

Reviewer: ```suggestion
-  The space's avatar if the file is in a space or the user's avatar if the file is in Personal documents 
```

-  The file's path
-  The file's name which is editable

Summary field
--------------

The summary text field allows you to add a description of the changes you made on the document.

.. note:: The summary field is not mandatory, if you input it it will be also added as summary for the document's versions 
          list in Documents application.
          
Versions list
--------------

A list of the latest three versions appears in the drawer. Each version contains these information:

-  The editor's avatar
-  The changes timestamp, when hovering it a tooltip appears with the real edit date
-  The edit summary which could be:

   -  The summary inputted when saving the version
   -  The text message "File has been updated" if the document's editor did not input a summary while saving the versions
   -  The text message "File has been renamed to newname.extension" if the document's editor has renamed the file
   

> Clicking on the ``See more`` button redirects you to the Documents application. YOu can then open Versions list by clicking 

Reviewer: ```suggestion
Clicking on the ``See more`` button redirects you to the Documents application. You can then open Versions list by clicking 
```

> on Versions button on the toolbar.

Reviewer: ```suggestion
on ``Versions`` button on the toolbar.
```

|image9|

Yellow bar
~~~~~~~~~~~~

> When a document is opened in your stream, either in ``Documents`` application or in the activity stream, has been edited by

Reviewer: ```suggestion
When a document visible in your stream, either in ``Documents`` application or in the activity stream, has been edited by
```
